services/activitylog/pkg/service/service.go

```diff
@@ -164,17 +192,42 @@ func (a *ActivitylogService) Activities(rid *provider.ResourceId) ([]Activity, e
 	}
 
 	if len(records) == 0 {
-		return []Activity{}, nil
+		return []RawActivity{}, nil
 	}
 
-	var activities []Activity
+	var activities []RawActivity
 	if err := json.Unmarshal(records[0].Value, &activities); err != nil {
 		return nil, fmt.Errorf("could not unmarshal activities: %w", err)
 	}
 
 	return activities, nil
 }
 
+// RemoveActivities removes the activities from the given resource
```

Comment:
Shouldn't we add a mutex to the ActivitylogService structure? 
We are consuming and writing the events in Run() and at the same time we are doing threads unsafe `read` `write` sequence in a RemoveActivities

Reply:
Now it is deadlock
You locked for the r/w `a.lock.Lock()` in RemoveActivities and after that try to `a.lock.RLock()` in Activities with waits forever

Reply:
The proper solution could be to split the `Activities` method and use the private in case if we have to read write
```
func (a *ActivitylogService) Activities(rid *provider.ResourceId) ([]RawActivity, error) {
       	a.lock.RLock()
	defer a.lock.RUnlock()
        return s.activities(rid)
}

func (a *ActivitylogService) activities(rid *provider.ResourceId) ([]RawActivity, error) {
	resourceID := storagespace.FormatResourceID(*rid)

        records, err := a.store.Read(resourceID)
        if err != nil && err != microstore.ErrNotFound {
		return nil, fmt.Errorf("could not read activities: %w", err)
	}

	if len(records) == 0 {
		return []RawActivity{}, nil
	}

	var activities []RawActivity
	if err := json.Unmarshal(records[0].Value, &activities); err != nil {
		return nil, fmt.Errorf("could not unmarshal activities: %w", err)
	}

	return activities, nil
}
```

Reply:
Looks good for me